libazureinit/src/distro.rs

```diff
@@ -26,10 +27,78 @@ impl Distribution for Distributions {
         password: &str,
     ) -> Result<i32, String> {
         match self {
+            Distributions::AzureLinux => {
+                let mut home_path = "/home/".to_string();
+                home_path.push_str(username);
+                println!("Creating user in AzureLinux: {}", username);
+                match Command::new("useradd")
+                    .arg(username)
+                    .arg("--comment")
+                    .arg(
+                      "Provisioning agent created this user based on username provided in IMDS",
+                    )
+                    .arg("-U")
+                    .arg("--groups")
+                    .arg("wheel,sudo")
+                    .arg("-d")
+                    .arg(home_path.clone())
+                    .arg("-m")
+                    .status(){
+                        Ok(_)=>(),
+                        Err(err) => return Err(err.to_string()),
+                    };
+
+                if password.is_empty() {
+                    match Command::new("passwd")
+                        .arg("-d")
+                        .arg(username)
+                        .status()
+                    {
+                        Ok(status_code) => {
+                            if !status_code.success() {
+                                return Err("Failed to create user".to_string());
+                            }
+                        }
+                        Err(err) => return Err(err.to_string()),
+                    };
+                    println!("User created without password")
+                } else {
+                    let input = format!("{}:{}", username, password);
+                    
+                    let mut output = Command::new("chpasswd")
```

Comment:
Sorry, I am not an expert of Azure Linux, but is there a specific reason why you need to do password-based authentication?

Previously we had the password authentication in the code base, but the PR https://github.com/Azure/azure-init/pull/57 removed that part mainly for security. Though there still seems to be cases where password authentication would be necessary. I am just curious what the situation of Azure Linux would be.

Reply:
No actually, I believe we do want to eliminate password auth, this was a hold over from me re-implementing the debian logic of an older branch 👍

Thank you for the catch